deepchecks/nlp/checks/data_integrity/property_label_correlation.py

```diff
@@ -92,8 +92,18 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         DeepchecksValueError
             If the object is not a Dataset instance with a label.
         """
-        text_data = context.get_data_by_kind(dataset_kind).sample(self.n_samples, random_state=context.random_state)
+        type_name = type(self).__name__
 
+        if context.task_type is TaskType.TOKEN_CLASSIFICATION:
+            task_type_name = TaskType.TOKEN_CLASSIFICATION.value
+            raise DeepchecksNotSupportedError(f'"{type_name}" is not suited for the "{task_type_name}" tasks')
+
+        text_data = context.get_data_by_kind(dataset_kind)
+
+        if text_data.is_multi_label_classification():
```

Comment:
same as above - maybe can even be under the same function?

Reply:
fixed,
I leave them seperate